addons/lunch/static/src/js/lunch_kanban_model.js

```diff
@@ -74,9 +74,18 @@ var LunchKanbanModel = KanbanModel.extend({
             },
         });
     },
+    _setUserLocation: function () {
+        return this._rpc({
+            route: '/lunch/user_location_set',
+            params: {
+                location_id: this.locationId,
+                user_id: this.userId,
+            },
+        });
+    },
     _updateLocation: function (locationId) {
         this.locationId = locationId;
-        return $.when();
+        return this._setUserLocation();
```

Comment:
wouldn't it make more sense to update the widget's location only on success?
```javascript
_setUserLocation: function(locationId) {
    return this._rpc({
            route: '/lunch/user_location_set',
            params: {
                location_id: locationId,
                user_id: this.userId,
            }, 
    });
},
_updateLocation: function(locationId) {
    var self = this;
    return this._setUserLocation(locationId).then(function () {
        self.locationId = locationId;
    });
}
```
it just seems safer to update the ui only upon a successful rpc from my PoV

Reply:
Yes it seems better, I'll update the PR